fix(deployment): bonita portal as remote client of the bonita engine

tbouffard opened the pull request:
The previous way of document this was mixed with how to configure custom
code being client of the Bonita Engine.
In addition, provide explanation for both the Tomcat and Wildfly bundles

tbouffard committed:
fix(deployment): bonita portal as remote client of the bonita engine

The previous way of document this was mixed with how to configure custom
code being client of the Bonita Engine.
In addition, provide explanation for both the Tomcat and Wildfly bundles

tbouffard commented:
**Important**: modifications must be applied to the 7.9 backport in the Tomcat bundle Bonita Portal configuration as we have switched from Bitronix to Arjuna/Narayana
The introduction must also be changed as there is no more `deploy` bundle

ecuriotto commented on md/configure-client-of-bonita-bpm-engine.md:
```diff
@@ -4,6 +4,10 @@ If you want your application to use the engine API of a remote engine, your appl
 
 The engine (the *server*) must also be configured to accept the connection.
 
+**Note**: the Bonita Portal can be configured to be used as a Client of the Bonita Engine. See the [Alternate Bonita Deployment](two-main-types-of-deployment.md)
```
Is it just the Bonita Portal or all the web layer?

tbouffard replied:
In the documentation, we only talk about the Portal.

ecuriotto commented on md/two-main-types-of-deployment.md:
````diff
@@ -50,7 +45,150 @@ But you can still use the pre packaged Tomcat bundles or the `bonita.war` webapp
+       goto exit
+   )
+```
+
````
Should we add in this paragraph to remove the setup folder?

tbouffard replied:
I document to use the setup tool to configure the Bonita Portal bundle but for now this is only for the setenv file
I have to check how we document the alternative authentication configuration which is a mix between server side and Bonita configuration (which requires configuration in the setup tool only used with the Engine part here)

tbouffard replied:
done with an additional commit

ecuriotto commented on md/two-main-types-of-deployment.md:
```diff
@@ -50,7 +45,150 @@ But you can still use the pre packaged Tomcat bundles or the `bonita.war` webapp
+
+Apply the same changes as for the Tomcat bundle to disable Engine resources
+
+
```
In the "Configuring the Bonita Stack" chapter there's no mention to changes in the server with the engine. Is it correct?

tbouffard replied:
Yes, that's correct
I will add a section about this to make this explicit

ecuriotto commented on md/two-main-types-of-deployment.md:
```diff
@@ -26,20 +26,15 @@ It is fast because the Bonita Portal and the Bonita Engine run on the same JVM a
-
 ### Accessible through HTTP
 
 ![deploy2](images/images-6_0/poss_deploy2.png)
```
It should be good to add to the picture a reference to the two application servers and then use it in the chapter. If we use "Bonita portal", this term should appear in the picture

tbouffard replied:
I don't understand `portal` is already present in all pictures of this page

![image](https://user-images.githubusercontent.com/27200110/59117451-0a040200-894e-11e9-8a4d-3759782b09e4.png)

ecuriotto reviewed (commented):
Very good, following these instructions I could setup a working environment with client and engine in separate servers

tbouffard committed:
review feedback: remove usage of the setup tool on Portal side

This highly simplify the installation procedure and make it clear that
it is not required by the Portal

tbouffard commented:
Merging to propose the solution that has been validated by @ecuriotto 
Feel free to open new pull request to improve this documentation
